plugins.py

```diff
@@ -0,0 +1,83 @@
+import time
+import datetime
+import deltachat
+
+
+class ReceivePlugin:
+    name = "test account"
+
+    @staticmethod
+    @deltachat.account_hookimpl
+    def ac_incoming_message(message):
+        received = time.time()
+        message.create_chat()
+        if not message.chat.can_send():
+            # if it's a device message or mailing list, we don't need to look at it.
+            return
+        sender = message.get_sender_contact().addr
+        receiver = message.account.get_self_contact().addr
+        msgcontent = parse_msg(message.text)
+        firsttravel = msgcontent.get("testduration")
+        secondtravel = received - msgcontent.get("begin")
+        iam = msgcontent.get("sender")
+        if message.chat.is_group():
+            if iam == "spider":
+                print("%s: joined group chat %s after %.1f seconds" % (receiver, message.chat.get_name(), secondtravel))
+                message.chat.send_text("Sender: %s\nBegin: %s" % (receiver, str(time.time())))
+                return  # for now it's enough to just accept the group chat
+            else:
+                return  # when messages by others arrived is checked later by the main thread
```

Comment:
The control flow in this plugin/method is super weird.

During the tests, each test account receives at least 3 messages:
- the single message from spider (becomes large file soon, see #8)
- the group add message from spider
- one message per other group member

These messages need to be distinguished and treated differently by this function. And right now, this is a mess. The problem is, all kinds of messages could arrive here. How to weed out the useful ones?

Reply:
I mean, right now it works. But it's messy and needs refactoring.

Reply:
fixed in 20ce15b